pkg/flow/flow_components.go

```diff
@@ -0,0 +1,240 @@
+package flow
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/flow/internal/controller"
+	"github.com/grafana/agent/pkg/flow/internal/dag"
+	"github.com/grafana/agent/pkg/river/encoding/riverjson"
+)
+
+// ComponentID is a globally unique identifier for a component.
+type ComponentID struct {
+	ModuleID string // Unique ID of the module that the component is running in.
+	LocalID  string // Local ID of the component, unique to the module it is running in.
+}
+
+// String returns the "<ModuleID>/<LocalID>" string representation of the id.
+func (id ComponentID) String() string {
+	if id.ModuleID == "" {
+		return id.LocalID
+	}
+	return id.ModuleID + "/" + id.LocalID
+}
+
+type ComponentDetail struct {
+	// Component is the instance of the component. Component will be nil if the
+	// component exists in the graph but has not successfully evaluated yet.
+	Component component.Component
```

Comment:
Is Component used anywhere?

Reply:
Not at the moment. It will play a critical part in my design for services though.